docs/how_to/compose_quantum_circuits.rst

```diff
@@ -0,0 +1,184 @@
+#####################
+Join quantum circuits
+#####################
+
+This guide shows you how to combine different :class:`~.QuantumCircuit` objects.
+
+Create the circuits
+===================
+
+Let's first create two circuits, which will be joined together.
+
+.. testcode::
+
+    from qiskit import QuantumCircuit
+
+    qc1 = QuantumCircuit(2,1)
+    qc1.h(0)
+    qc1.cx(0,1)
+    qc1.measure(1,0)
+
+    qc2 = QuantumCircuit(4,2)
+    qc2.y(0)
+    qc2.x(1)
+    qc2.cx(0,3)
+    qc2.measure(3,1)
+
+    print(qc1.draw()) 
+    print(qc2.draw())
+
+.. testoutput::
+    :options: +NORMALIZE_WHITESPACE
+
+         ┌───┐        
+    q_0: ┤ H ├──■─────
+         └───┘┌─┴─┐┌─┐
+    q_1: ─────┤ X ├┤M├
+              └───┘└╥┘
+    c: 1/═══════════╩═
+                    0 
+         ┌───┐        
+    q_0: ┤ Y ├──■─────
+         ├───┤  │     
+    q_1: ┤ X ├──┼─────
+         └───┘  │     
+    q_2: ───────┼─────
+              ┌─┴─┐┌─┐
+    q_3: ─────┤ X ├┤M├
+              └───┘└╥┘
+    c: 2/═══════════╩═
+                    1 
+
+Join the circuits
+=================
+
+You can join the circuits together using these two methods:
+
+1. Using the :meth:`~.QuantumCircuit.compose` method
+2. Using the :meth:`~.QuantumCircuit.append` method
+
+.. warning::
+
+     The ``combine`` and ``extend`` methods have been deprecated in Qiskit Terra 0.17 and removed in 0.23. These methods are replaced by the :meth:`~.QuantumCircuit.compose` method which is more powerful. The removal of ``extend`` also means that the ``+`` and ``+=`` operators are no longer defined for :class:`~.QuantumCircuit`. Instead, you can use the ``&`` and ``&=`` operators respectively, which use :meth:`~.QuantumCircuit.compose`.
+
+For both methods, if the two circuits being combined have different sizes, the method needs to be called in the circuit that is bigger (more qubits and more classical bits). 
+
+Using the :meth:`~.QuantumCircuit.compose` method
+-------------------------------------------------
+
+In order to join two circuits with :meth:`~.QuantumCircuit.compose`, you only have to specify the circuit you want to insert. That way the qubits and bits of the smaller circuit will be included into the first qubits and bits of the bigger one in the original order they had. 
+
+By default, :meth:`~.QuantumCircuit.compose` does not modify the original circuit to which it is applied but returns a new joint circuit object. This can be changed by setting the ``inplace`` argument to ``True``.
```

Comment:
I think it would be good to put this sentence in a note box or a in brackets, as the following code example doesn't actually show how to use the inplace argument

Reply:
alternatively it might be nice to add a short code snippet showing the use of the inplace arg